### PR TITLE
Api/assignments

### PR DESCRIPTION
**Pull request by Padraic-0**
Completed delete and patch

### CODE REVIEW

File: src/api/assignments.ts [hunk 1]
```diff
@@ -32,9 +32,32 @@ router.get("/:id/submissions", (req, res) => {
...
+        const user = await User.findById(req.userId!);
+        return user.role === "admin" || user.id == instructorId;
+    }
+}), async(req, res) => {
   const assignmentId = req.params.id;
-  // TODO: Implement
+  const assignment = await Assignment.findById(assignmentId);
+  try {
+    const deleted = await assignment.delete()
+    res.status(200).json({message: "Assignment Deleted"});
```
Comment (theusaf) [inline src/api/assignments.ts:57]: Delete should be status `204` with no content.

File: src/api/assignments.ts [hunk 2]
```diff
@@ -32,9 +32,32 @@ router.get("/:id/submissions", (req, res) => {
...
+  } catch (error) {
+    res.status(400).json({error: "Assignment not Deleted"});
```
Comment (theusaf) [inline src/api/assignments.ts:59]:
```suggestion
    res.status(500).json({error: "Assignment not Deleted"});
```

File: src/api/assignments.ts [hunk 3]
```diff
@@ -32,9 +32,32 @@ router.get("/:id/submissions", (req, res) => {
...
+  }
   res.status(204).send();
```
Comment (theusaf) [inline src/api/assignments.ts:61]:
```suggestion
```

File: src/api/assignments.ts [hunk 4]
```diff
@@ -56,12 +79,24 @@ router.patch("/:id", allowedInBody(["title", "points", "due"]), requireAuthentic
...
+    const assignmentData = req.body;
+    const savedAssignment = await Assignment.findById(assignmentId);
+
+    try {
+        Object.keys(assignmentData).forEach( key => {
+        (savedAssignment as any)[key] = assignmentData[key]
+        });
+
+        const assignment = savedAssignment.save();
+        res.status(200).json({message: "Updated assignment", assignment: assignment});
```
Comment (theusaf) [inline src/api/assignments.ts:95]:
```suggestion
        res.status(200).json(assignment.toJSON());
```

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by Padraic-0 — COMMENTED** [review]
Looks good

File: src/api/assignments.ts (hunk 1 above)
Comment (theusaf) [reply src/api/assignments.ts:57]:
```suggestion
    res.status(204).send();
```

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by Padraic-0 — COMMENTED** [review]
Looks good

**Review by Padraic-0 — COMMENTED** [review]
Looks good

**Comment by Padraic-0** [comment]
Great

File: src/api/assignments.ts [hunk 5]
```diff
@@ -32,10 +50,32 @@ router.get("/:id/submissions", (req, res) => {
...
+        return user.role === "admin" || user.id == instructorId;
+    }
+}), async(req, res) => {
   const assignmentId = req.params.id;
-  // TODO: Implement
-  res.status(204).send();
+  const assignment = await Assignment.findById(assignmentId);
+  try {
+    const deleted = await assignment.delete()
+    res.status(204);
```
Comment (theusaf) [inline src/api/assignments.ts:75]:
```suggestion
    res.status(204).send();
```

**Review by theusaf — APPROVED** [review]

**Pull request merged**